src/librustc/dep_graph/dep_node.rs

```diff
@@ -148,6 +148,10 @@ pub enum DepNode<D: Clone + Debug> {
     // For proj. cache, we just keep a list of all def-ids, since it is
     // not a hotspot.
     ProjectionCache { def_ids: Vec<D> },
+
+    // Depnodes for MetaData
```

Comment:
Heh, you can leave the comment off - the reason we need these is *because* `MetaData` can't be used, as it's reserved for cross-crate queries, e.g. `ItemSignature(def_id)` will eventually depend on either `Hir(def_id)` iff `def_id.is_local()` or `MetaData(def_id)` otherwise.